src/client.js

```diff
@@ -137,23 +104,43 @@ class Client extends events.EventEmitter {
     }
 
     if (this.running) {
-      const newConn = new Connection(this);
-
-      // don't actually start connecting until at lesat minReconnectionInterval has passed
-      const earliestConnectionTime = this.lastConnectionTime + this._minReconnectionInterval;
-      const now = new Date().getTime();
-      setTimeout(() => {
-        this.lastConnectionTime = new Date().getTime();
-        newConn.connect();
-      }, now < earliestConnectionTime ? earliestConnectionTime - now : 0);
-
-      newConn.once('connected', () => {
-        this.emit('connected', newConn);
-      });
-      newConn.once('finished', () => {
-        this.connections = this.connections.filter(conn => conn !== newConn);
-      });
-      this.connections.unshift(newConn);
+      // If recycle is called due to recycleAt then refresh the normal recycle interval
+      if (this.recycleAt && this.recycleAt - new Date() === 0) {
```

Comment:
This will almost never be true. Time will pass between when `recycle` is called and you construct your new `Date`. I think you actually shouldn't make this check anyway. The old way with `setTimeout` seems to make more sense to me.

Reply:
There's still more work to do in this function to line up with what @djmitche was saying in [this comment](https://github.com/taskcluster/taskcluster-lib-pulse/pull/7#discussion_r205481788). For instance `Connection` should not take a retirement delay at all, right? That is entirely managed by the client.

Reply:
Ah that's my fault. @djmitche  wanted to drop the callback effect with `setTimeout` and seperately handle that logic. I think something like `this.recycleAt - new Date() > 5 && this.recycleAt - new Date() < 15` can make sense. Do same in  `this.claimedCredentials` too.

Reply:
@djmitche  I wanted to how's the above mentioned logic

Reply:
I agree -- comparing a time for equality will never be true.

Reply:
@djmitche equality will never be true but  can we do something like `this.recycleAt- new Date` is in interval [-15,15]ms

Reply:
That's still assuming things happen within 15ms, and when they don't it might cause a failure.  It's best to avoid this kind of situation entirely.  I think my comments in the review are a way to do so?